gnovm/pkg/gnolang/nodes.go

```diff
@@ -1116,7 +1116,12 @@ func ReadMemPackage(dir string, pkgPath string) *std.MemPackage {
 		".gno",
 	}
 	list := make([]string, 0, len(files))
+	isRealm := strings.HasPrefix(pkgPath, "gno.land/r/")
 	for _, file := range files {
+		if isRealm && file.IsDir() {
+			// currently realm packages does not support subpackages https://github.com/gnolang/gno/issues/1041
+			panic("not supported yet")
+		}
 		if file.IsDir() ||
```

Comment:
If you look at this check, we are already ignoring subdirectories. 

I don't think panicking is a good idea.

There are many reasons why I want to have subdirectories in realms:

- Nested Independent Realms:

    Imagine I have two realms: `examples/gno.land/p/demo/foo` and `examples/gno.land/p/demo/foo/bar`. 
Right now, I can publish both realms in two transactions. However, with these changes, I won't be able to publish `foo` since it contains `bar`.

- Local Directories:

    Imagine I have directories like `/tests` or `/docs` that I don't want to publish but want to keep locally. I need to remove them to be able to publish realm.

Reply:
Hello @harry-hov  thanks for your review 👍

> I don't think panicking is a good idea.
I think that is just what the issue ask us to do (maybe it not relevant anymore ?)

> There are many reasons why I want to have subdirectories in realms:
> 
> Nested Independent Realms:
> 
> Imagine I have two realms: examples/gno.land/p/demo/foo and examples/gno.land/p/demo/foo/bar.
> Right now, I can publish both realms in two transactions. However, with these changes, I won't be able to publish foo since it contains bar.

This should apply only for **r** Realms not for **p** Realms so you will be able to keep using the subdirectory on `examples/gno.land/p/demo/foo` and `examples/gno.land/p/demo/foo/bar`

> Local Directories:
> 
> Imagine I have directories like /tests or /docs that I don't want to publish but want to keep locally. I need to remove them to be able to publish realm.

I think you have a really good point here, maybe we will need to only block the subdirectories that contains a .gno file ?

Reply:
@zivkovicmilos WDYT about the panic comment? The initial issue has panic in the scope.